packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'os';
+import {
+  DebugSession,
+  OutputEvent,
+  Source,
+  StoppedEvent,
+  Thread,
+  ThreadEvent
+} from 'vscode-debugadapter';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { LogFile } from '../core/logFile';
+import { nls } from '../messages';
+
+export interface LaunchRequestArguments
+  extends DebugProtocol.LaunchRequestArguments {
+  logFile: string;
+  stopOnEntry: boolean | true;
+  traceLog: boolean | true;
+}
+
+export class ApexReplayDebug extends DebugSession {
+  public static THREAD_ID = 1;
```

Comment:
Unlike the Apex Interactive Debugger, we don't allow debugging multiple threads, so just hardcode this ID.

Reply:
I think this is true (or at least for all intents and purposes it might not matter much).

I'm curious when we have a situation like https://help.salesforce.com/articleView?id=code_debug_log_details.htm&type=5 (more description on the different code_units) are also there

```
EXECUTION_STARTED
CODE_UNIT_STARTED|[EXTERNAL]execute_anonymous_apex
CODE_UNIT_STARTED|[EXTERNAL]MyTrigger on Account trigger event BeforeInsert for [new]
CODE_UNIT_FINISHED <-- The trigger ends
CODE_UNIT_FINISHED <-- The executeAnonymous ends
EXECUTION_FINISHED
```

This seems to be under two different CODE_UNITS. Are we just going to lump them into one thread? AFAIC, lumping them under one thread should not cause any issues since the log has already serialized them linearly so we can just step through them (there is no weird interleaving).

Reply:
I haven't seen any interleaving, so I was going to treat CODE_UNIT_STARTED/FINISHED like entering and leaving a stackframe.